R/object_overwrite_linter.R

```diff
@@ -102,6 +102,8 @@ object_overwrite_linter <- function(
 
     assigned_exprs <- xml_find_all(xml, xpath_assignments)
     assigned_symbols <- get_r_string(assigned_exprs, "SYMBOL")
+    is_quoted <- startsWith(assigned_symbols, "`")
```

Comment:
this won't catch assignments with ' or "

Reply:
fixed. Won't bother with raw strings though.

Reply:
Those weren't caught by the XPath before, because they are `//STR_CONST`. `get_r_string()` handles removal of `"`, `'` and even `r"()"`. I added tests for the first two cases.

Reply:
> even r"()"

The whole original point of `get_r_string()` was to account for that! :smile: 

Almost surely there's at least one other linter using the same logic to find assigned symbols, probably that should be kept in sync. But also I'm sure there's dozens of such cases in our code base by now of XPaths with 99% overlap that should be relegated to helpers.